# FUSARIUM-ID v3.0 for QIIME 2 version 2024.2

Naive Bayes Classifiers for [FUSARIUM-ID v3.0](https://github.com/fusariumid/fusariumid/releases/tag/v.3.0) trained for QIIME 2 version [2024.2](https://docs.qiime2.org/2024.2/install/).

I am not affiliated to the FUSARIUM-ID team. Classifiers are provided _as is_, without any guarantees or warranties.

If you use these classifiers, you should cite the following papers:

- **Workflow methods:**
> Bokulich, N. A., Kaehler, B. D., Rideout, J. R., Dillon, M., Bolyen, E., Knight, R., Huttley, G. A., & Caporaso, J. G. (2018). Optimizing taxonomic classification of marker-gene amplicon sequences with QIIME 2’s q2-feature-classifier plugin. Microbiome, 6(1), 90. https://doi.org/10.1186/s40168-018-0470-z
>
> Bolyen, E., Rideout, J. R., Dillon, M. R., Bokulich, N. A., Abnet, C. C., Al-Ghalith, G. A., Alexander, H., Alm, E. J., Arumugam, M., Asnicar, F., Bai, Y., Bisanz, J. E., Bittinger, K., Brejnrod, A., Brislawn, C. J., Brown, C. T., Callahan, B. J., Caraballo-Rodríguez, A. M., Chase, J., … Caporaso, J. G. (2019). Reproducible, interactive, scalable and extensible microbiome data science using QIIME 2. Nature Biotechnology, 37(8), 852–857. https://doi.org/10.1038/s41587-019-0209-9
>
> Pedregosa, F., Varoquaux, G., Gramfort, A., Michel, V., Thirion, B., Grisel, O., Blondel, M., Prettenhofer, P., Weiss, R., Dubourg, V., Vanderplas, J., Passos, A., Cournapeau, D., Brucher, M., Perrot, M., & Duchesnay, É. (2011). Scikit-learn: Machine learning in Python. Journal of Machine Learning Research, 12(Oct), 2825–2830.
>
> Wes McKinney. (2010). Data Structures for Statistical Computing in Python . In S. van der Walt & Jarrod Millman (Eds.), Proceedings of the 9th Python in Science Conference (pp. 51–56).

- **FUSARIUM-ID v3.0:**
> Torres-Cruz, T. J., Whitaker, B. K., Proctor, R. H., Broders, K., Laraba, I., Kim, H. S., ... & Geiser, D. M. (2022). FUSARIUM-ID v. 3.0: An updated, downloadable resource for Fusarium species identification. Plant disease, 106(6), 1610-1616.

